Testing - ensureArray create test

YPAzevedo opened the pull request:
Fixes #8

YPAzevedo committed:
chore: add test to ensureArray.ts

YPAzevedo committed:
chore: fix var name on test

nobrayner commented on test/ensureArray.test.ts:
```diff
@@ -0,0 +1,9 @@
+import { ensureArray } from '../src/ensureArray';
+
+it('any value passed should return array', () => {
+  const arrayWithTestSting = ensureArray(['test']);
+  const testString = ensureArray('test');
+
+  expect(arrayWithTestSting).toEqual(['test']);
+  expect(testString).toEqual(['test']);
+});
```
You are testing two different cases here:
1. It should return an array with the given value
2. It return the given value if it is an array

I would advise separating these into two different tests - that way the intent is much clearer on what is happening :)

YPAzevedo replied:
So it should the two `it`, right? also in the second one where i test for `It should return an array with the given value` should i add more types to test or only one thats different from an array, enough? 
```js
it('should return an array with the given value', () => {
  const testString = ensureArray('test');
  const testNumber = ensureArray(0);
  const testBoolean = ensureArray(true);

  expect(testString).toEqual(['test']);
  expect(testNumber).toEqual([0]);
  expect(testBoolean).toEqual([true]);
});
```
or only:
```js
it('should return an array with the given value', () => {
  const testString = ensureArray('test');

  expect(testString).toEqual(['test']);
});
```

nobrayner replied:
Hmm... Good question! Based on the fact that the `ensureArray` method is based on `any` type... I would check:
* A primitive
* A function
* An object

Between those three, I'd say you'd have it covered

YPAzevedo replied:
@nobrayner Yeah, makes sense. Updated the test add the assertions for those 3 different value types. let me know if need to change something else 👍

nobrayner reviewed (changes requested):
Awesome! Thanks for getting this started! :smile:

YPAzevedo committed:
chore: separate test add more assertions

nobrayner reviewed (approved):
Now we just need to setup tests to run on CI :sweat_smile:

YPAzevedo committed:
chore: improve test for ensureArray.ts

YPAzevedo committed:
chore: fix should typo on test

YPAzevedo commented:
@nobrayner updated the test, added `toBe` on the first one since if it's an array it should be the same *value* it was passed in, so i added more test cases.

nobrayner commented on test/ensureArray.test.ts:
```diff
@@ -0,0 +1,29 @@
+  const ensureArrayGetFourtyTwoArray = ensureArray(getFourtyTwoArray);
+  const ensureArrayObjFourtyTwoArray = ensureArray(objFourtyTwoArray);
+
+  expect(ensureArrayFourtyTwoArray).toBe(fourtyTwoArray);
+  expect(ensureArrayGetFourtyTwoArray).toBe(getFourtyTwoArray);
+  expect(ensureArrayObjFourtyTwoArray).toBe(objFourtyTwoArray);
```
Sweet! Good catch :+1:

nobrayner commented:
Merging this, as fixing CI to run on PRs won't cause CI to run here.
